versions.sh

```diff
@@ -86,12 +86,39 @@ for version in "${versions[@]}"; do
 		jq <<<"$json" -c --argjson doc "$doc" '
 			.[env.version] = {
 				version: env.fullVersion,
-				variants: [ "apache-buster", "fpm-buster", "fpm-alpine3.12" ],
+				variants: [
+					# arrays of arrays to control ordering
+					if [ "7", "8.9", "9.1" ] | index(env.version) then
+						[
+							[ "buster" ],
+							[ "alpine3.13" ]
+						]
+					elif env.version | startswith("9.") then
+						[
+							[ "buster" ],
+							[ "alpine3.14", "alpine3.13" ]
+						]
+					else
+						[
+							[ "bullseye", "buster" ],
+							[ "alpine3.14", "alpine3.13" ]
+						]
+					end
+					| . as $os
+					| [
+						[ "apache", "fpm" ],
+						[ "fpm" ]
+					] as $variant
+					| keys[]
+					| $variant[.][] as $var | $os[.][] | [ $var, . ] | join("-")
+				],
 				phpVersions: (
-					if [ "7", "8.9", "9.0"] | index(env.version) then
+					if [ "7", "8.9" ] | index(env.version) then
 						[ "7.4" ]
-					else
+					elif env.version | startswith("9.") then
 						[ "8.0", "7.4" ]
+					else
+						[ "8.0" ]
```

Comment:
Do you think we should add some links in here to why we're doing this? (like that one you found that said 8+ only)